app/models/tag.rb

```diff
@@ -108,6 +108,10 @@ def tagged_documents
     @_tagged_documents ||= TaggedDocuments.new(self)
   end
 
+  def tagged_document_for_base_path(base_path)
+    tagged_documents.find { |document| document.base_path == base_path }
```

Comment:
The docs suggest `find` and `detect` are synonyms - can you explain the difference?
http://ruby-doc.org/core-2.3.1/Enumerable.html